homeassistant/components/imap/coordinator.py

```diff
@@ -166,8 +165,13 @@ def _decode_payload(part: Message) -> str:
             Falls back to the raw content part if decoding fails.
```

Comment:
Side note: I think we should move out `_decode_payload` to the module level. It no longer needs any nested scope variables.

Reply:
https://github.com/home-assistant/core/pull/109665